### PR TITLE
✅(person) enhance person factory

### PR DESCRIPTION
**Pull request by sveetch**
## Purpose

Person factory lacked to implement portrait and resume, also it was 
not included in demonstration site creation.

## Proposal

This has been performed with portrait and resume field implementation to the 
Factory, adding test to cover factory and finally included Person batch 
creation inside 'create_demo_site' commandline.

### CODE REVIEW

File: apps/core/tests/utils.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+"""
+Usefull stuff for unittest
```
Comment (sampaccoud) [inline apps/core/tests/utils.py]: s/Usefull/Useful
s/unittest/testing  (technically not unit tests but integration tests)

File: apps/persons/factories.py [hunk 2]
```diff
@@ -43,5 +53,49 @@ def extended_object(self):
...
-            "en",
+            settings.LANGUAGE_CODE,
+            parent=self.parent,
         )
+
+    @factory.post_generation
+    # pylint: disable=unused-argument
+    def with_content(self, create, extracted, **kwargs):
+        """
+        Add content plugins displayed in the "maincontent" placeholder of the organization page:
```
Comment (sampaccoud) [inline apps/persons/factories.py]: Should be:
```
Add content plugins displayed in the specific placeholders of the person page
```

File: apps/persons/factories.py [hunk 3]
```diff
@@ -43,5 +53,49 @@ def extended_object(self):
...
+        - Picture plugin featuring a random portrait image,
+        - Text plugin featuring a long random description.
```
Comment (sampaccoud) [inline apps/persons/factories.py]: featuring the person's resume

File: apps/persons/factories.py [hunk 4]
```diff
@@ -43,5 +53,49 @@ def extended_object(self):
...
+            portrait = Image.objects.create(file=wrapped_portrait)
+            add_plugin(
+                language=language,
+                placeholder=portrait_placeholder,
+                plugin_type="PicturePlugin",
+                picture=portrait,
+                attributes={"alt": "portrait image"},
+            )
+
+            # Add a text plugin with a long random description
```
Comment (sampaccoud) [inline apps/persons/factories.py]: with the person's resume

File: apps/persons/tests/test_factories_person.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+            plugin_type="PicturePlugin"
+        )
+        self.assertIn(
+            "portrait",
+            os.path.basename(
+                portrait_plugin.djangocms_picture_picture.picture.file.name
+            ),
+        )
+
+        # The description plugin should contain paragraphs
```
Comment (sampaccoud) [inline apps/persons/tests/test_factories_person.py]: Should be: The resume plugin

File: apps/persons/tests/test_factories_person.py [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    """
+
+    def test_person_factory(self):
+        """
+        PersonFactoryTestCase should be able to generate plugins with realistic fake
+        data: portrait and resume.
+        """
+        person = PersonFactory(with_content=True)
+
+        # The banner plugin should point to one of our fixtures images
```
Comment (sampaccoud) [inline apps/persons/tests/test_factories_person.py]: Should be: The portrait plugin

**Comment by sveetch** [comment]
Ok i fixed some things from your feedbacks and rewrited my initial commit message.

File: apps/persons/factories.py [hunk 7]
```diff
@@ -43,5 +53,51 @@ def extended_object(self):
...
-            "en",
+            settings.LANGUAGE_CODE,
+            parent=self.parent,
         )
+
+    @factory.post_generation
+    # pylint: disable=unused-argument
+    def with_content(self, create, extracted, **kwargs):
+        """
+        Add content plugins displayed in the "maincontent" placeholder of the person
```
Comment (sampaccoud) [inline apps/persons/factories.py]: s/"maincontent" placeholder/"portrait" and "resume" placeholders

**Pull request merged**